oslo/transformers/tasks/data_bart_pretraining.py

```diff
@@ -1,92 +1,132 @@
+            pattern = r"(?<=\w[.!?])(?<![A-Z][.!?])(?<![0-9][.!?])"
+            splited_sents = [
+                self._tokenizer.bos_token 
+                + f'{self._tokenizer.pad_token}'.join(
+                    [
+                        sp_text.strip() 
+                        for sp_text in re.split(pattern, text) if len(sp_text) > 1 # Conditional statement for the last sentence
+                    ]
+                ) 
+                + self._tokenizer.eos_token 
+                for text in example['text']
+            ]
+            
+            tokenized_sents = self._tokenizer(
+                    splited_sents,
+                    padding=False,
+                    truncation=False,
+                    add_special_tokens=False,
+                    return_attention_mask=False,
+                    return_special_tokens_mask=False,
+                    verbose=False,
+            )['input_ids']
+            
+            return {'input_ids':tokenized_sents}
+        
+        split_tokenized_dataset =  dataset.map(
+            split_sentence_add_token,
+            batched = True,
+            num_proc = 2, #  number of multiprocessor
+            remove_columns=column_names 
+        )
+  
+        def group_text(examples):
+            self._buffer.extend(list(chain.from_iterable(examples['input_ids']))) # combine input_ids dataset 
+            total_length = len(self._buffer)
+            quotient, remain = divmod(total_length, self._max_seq_length)                            
+            output = [ self._buffer[i : i + self._max_seq_length] 
+                    for i in range(0, quotient * self._max_seq_length, self._max_seq_length)
+                    ] 
+            self._buffer = self._buffer[quotient * self._max_seq_length:]
+            return {'input_ids':output}
+        
+        group_dataset = split_tokenized_dataset.map(
+            group_text,
+            batched=True,
+            num_proc=2
+        )
+        
+        return group_dataset
+        
 
 
 class DataCollatorForBartPretraining(object):
     """
     Processing training examples to mini-batch for Bart (text_infilling, sentence_permutation).
+    processor:
+        pass
+    mask_ratio (:obj:`float`):
+        The probability with which to (randomly) mask tokens in the input
+    poisson_lambda (:obj:`float`):
+        Mean parameter of Poisson distribution used to generate span-lengths to be masked
+    permute_sentence_ratio (:obj:`float`):
+        Ratio of sentences to be permuted in each document
+    decoder_start_token_id: (:obj:`int):
+        The decoder start token id of the model
+    label_pad_token_id:
+        pass 
+    
     """
 
     def __init__(
         self,
         processor: ProcessorForBartPretraining,
-        mlm_probability: float = 0.15,
-        possion_lambda: float = 3.0,
-        permute_sentence: bool = True,
-        label_pad_token_id: int = -100,
+        mask_ratio: float = 0.3,
```

Comment:
variable name is needed to check

Reply:
These words (mask_ratio, permute_sentence_ratio, max_seq_length) are used in the huggingface.